framework/src/NormalisedSumPDF.cpp

```diff
@@ -295,119 +308,85 @@ double NormalisedSumPDF::Normalisation( DataPoint* NewDataPoint, PhaseSpaceBound
 {
 	(void)NewDataPoint;
 	(void)NewBoundary;
+	// Calculate and cache the integrals
+	for(DataPoint* pdp: integrationBoundary->GetDiscreteCombinations())
```

Comment:
The `RapidFitIntegrator` class does loop over discrete combinations (e.g. [here](https://github.com/gcowan/RapidFit/blob/master/framework/src/RapidFitIntegrator.cpp#L1112)), so I will remove this loop.

(it's a [range-based for](http://en.cppreference.com/w/cpp/language/range-for) from C++11)